app/controllers/concerns/noid_validator.rb

```diff
@@ -0,0 +1,36 @@
+# Copyright 2011-2023, The Trustees of Indiana University and Northwestern
+#   University.  Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+#   under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+#   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#   specific language governing permissions and limitations under the License.
+# ---  END LICENSE_HEADER BLOCK  ---
+
+module NoidValidator
+  extend ActiveSupport::Concern
+
+  include Identifier
+
+  included do
+    before_action :validate_noid, only: [:show]
+  end
+
+  private
+
+    def validate_noid
+      return if noid_service.valid?(params[:id])
+
+      # Strip all non-alphanumeric characters from passed in NOID
+      noid_id = params[:id]&.gsub(/[^A-Za-z0-9]/, '')
+
+      # If cleaned NOID is not valid raise error, otherwise redirect request.
+      raise ActiveFedora::ObjectNotFoundError unless noid_service.valid?(noid_id)
+      redirect_to(request.parameters.merge(id: noid_id))
```

Comment:
I was thinking this might be unsafe based upon the codeclimate warning and analysis:
>This is because params could contain :host => 'evilsite.com' which would redirect away from your site and to a malicious site.
>
>If the first argument to redirect_to is a hash, then adding :only_path => true will limit the redirect to the current host.

But since we're on Rails 7 there is a new default to watch for unsafe redirects and raise an error (https://edgeguides.rubyonrails.org/configuring.html#config-action-controller-raise-on-open-redirects) so I think we don't have to worry about the user manipulating the params. :slightly_smiling_face: